contracts/DutchAuction.sol

```diff
@@ -0,0 +1,176 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/common/ERC2981.sol";
+import "erc721a/contracts/ERC721A.sol";
+
+import "./IDutchAuction.sol";
+import "./ERC721M.sol";
+
+contract DutchAuction is IDutchAuction, ERC721M, ReentrancyGuard {
+    bool private immutable _refundable;
+    uint256 private _settledPriceInWei;
+    Config private _config;
```

Comment:
nit: rename to `DucthAuctionConfig` or similar

Reply:
I think it's fine, since the Config is specific to the IDutchAuction interface